rust/serving/src/app/jetstream_proxy.rs

```diff
@@ -33,34 +32,31 @@ use crate::config;
 //   "from_vertex": "a"
 // }
 
-const ID_HEADER_KEY: &str = "X-Numaflow-Id";
 const CALLBACK_URL_KEY: &str = "X-Numaflow-Callback-Url";
-
 const NUMAFLOW_RESP_ARRAY_LEN: &str = "Numaflow-Array-Len";
 const NUMAFLOW_RESP_ARRAY_IDX_LEN: &str = "Numaflow-Array-Index-Len";
 
-#[derive(Clone)]
 struct ProxyState<T> {
```

Comment:
don't the state need to be `Clone` anymore? make sure it is a cheap clone if there is a need.

Reply:
The router now takes `Arc<ProxyState>` as the state.